cluster/gce/gci/configure-helper.sh

```diff
@@ -1260,12 +1260,34 @@ EOF
   fi
 }
 
+# Create kubeconfig files for control plane components.
 function create-kubeconfig {
   local component=$1
   local token=$2
   echo "Creating kubeconfig file for component ${component}"
   mkdir -p "/etc/srv/kubernetes/${component}"
-  cat <<EOF >"/etc/srv/kubernetes/${component}/kubeconfig"
+  if [[ ${KUBECONFIG_USE_HOST_IP:-} == "true" ]] ; then
+    cat <<EOF >"/etc/srv/kubernetes/${component}/kubeconfig"
+apiVersion: v1
+kind: Config
+users:
+- name: ${component}
+  user:
+    token: ${token}
+clusters:
+- name: local
+  cluster:
+    insecure-skip-tls-verify: true
+    server: https://${KUBERNETES_MASTER_NAME:-localhost}:443
```

Comment:
nit: can we unify the if/else condition (e.g. add a local variable) if the rest of the kubeconfig remains the same? That would reduce the chance of diverging in future

Reply:
Cleaned up the logic to remove duplicate.